superset-frontend/src/explore/components/ExploreChartPanel.jsx

```diff
@@ -49,22 +52,124 @@ const propTypes = {
+
+const INITIAL_SIZES = [90, 10];
+const MIN_SIZES = [300, 50];
+
 const Styles = styled.div`
   background-color: ${({ theme }) => theme.colors.grayscale.light5};
   height: 100%;
   display: flex;
   flex-direction: column;
   align-items: stretch;
   align-content: stretch;
+  overflow: hidden;
+
   & > div:last-of-type {
     flex-basis: 100%;
   }
+
+  .gutter {
+    border-top: 1px solid ${({ theme }) => theme.colors.grayscale.light2};
+    border-bottom: 1px solid ${({ theme }) => theme.colors.grayscale.light2};
+    width: 3%;
+    margin: ${EXPLORE_GUTTER_MARGIN}px 47%;
+  }
+
+  .gutter.gutter-vertical {
+    cursor: row-resize;
+  }
+
+  .ant-collapse {
+    height: 100%;
+    background-color: ${({ theme }) => theme.colors.grayscale.light5};
+    .ant-collapse-item {
+      height: 100%;
+      border: 0;
+    }
+    .ant-collapse-content,
+    .ant-collapse-content-box {
+      height: 100%;
+    }
+    .ant-collapse-header {
+      padding-top: 0;
+      padding-bottom: 0;
+    }
+    .ant-tabs {
+      height: 100%;
+      .ant-tabs-nav {
+        padding-left: ${({ theme }) => theme.gridUnit * 5}px;
+        margin: 0;
+        background-color: ${({ theme }) => theme.colors.grayscale.light5};
+      }
+      .ant-tabs-content-holder {
+        overflow: hidden;
+        .ant-tabs-content {
+          height: 100%;
+        }
+      }
+    }
+  }
 `;
 
-class ExploreChartPanel extends React.PureComponent {
-  renderChart() {
-    const { chart } = this.props;
-    const headerHeight = this.props.standalone ? 0 : 80;
+const ExploreChartPanel = props => {
+  const panelHeadingRef = useRef(null);
+  const [headerHeight, setHeaderHeight] = useState(props.standalone ? 0 : 50);
+  const [splitSizes, setSplitSizes] = useState(INITIAL_SIZES);
+
+  const calcSectionHeight = percent => {
+    const containerHeight = parseInt(props.height, 10) - headerHeight - 30;
+    return (
+      (containerHeight * percent) / 100 -
+      (EXPLORE_GUTTER_HEIGHT / 2 + EXPLORE_GUTTER_MARGIN)
+    );
+  };
+
+  const [chartSectionHeight, setChartSectionHeight] = useState(
+    calcSectionHeight(INITIAL_SIZES[0]) - CHART_PANEL_PADDING,
+  );
+  const [tableSectionHeight, setTableSectionHeight] = useState(
+    calcSectionHeight(INITIAL_SIZES[1]),
+  );
+
+  useEffect(() => {
+    const calcHeaderSize = debounce(() => {
+      setHeaderHeight(
+        props.standalone ? 0 : panelHeadingRef?.current?.offsetHeight,
+      );
+    }, 100);
+    calcHeaderSize();
+    document.addEventListener('resize', calcHeaderSize);
+    return () => document.removeEventListener('resize', calcHeaderSize);
+  }, [props.standalone]);
+
+  const recalcPanelSizes = ([northPercent, southPercent]) => {
+    setChartSectionHeight(
+      calcSectionHeight(northPercent) - CHART_PANEL_PADDING,
+    );
+    setTableSectionHeight(calcSectionHeight(southPercent));
+  };
+
+  const onCollapseChange = openPanelName => {
+    const defaultSouthPaneOpenHeightPercent = 40;
```

Comment:
maybe make this a constant outside of the component?

Reply:
Done